packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,121 @@
 import type {
-  SchemaBuilder,
   Build,
   Context,
-  Plugin,
   Options,
+  Plugin,
+  SchemaBuilder,
 } from "graphile-build";
-import type { GraphQLInputFieldConfig, GraphQLFieldConfig } from "graphql";
+import {
+  GraphQLFieldConfig,
+  GraphQLInputFieldConfig,
+  GraphQLInputType,
+  GraphQLNonNull,
+  GraphQLType,
+} from "graphql";
+import { GraphQLList } from "graphql/type/definition";
+
+interface NullabilityOpts {
+  type?: boolean | boolean[];
+  args?: {
+    [argName: string]: boolean | boolean[];
+  };
+}
 
 interface ChangeNullabilityRules {
   [typeName: string]: {
-    [fieldName: string]: boolean;
+    [fieldName: string]: boolean | boolean[] | NullabilityOpts;
   };
 }
 
 export default function makeChangeNullabilityPlugin(
   rules: ChangeNullabilityRules
 ): Plugin {
   return (builder: SchemaBuilder, _options: Options) => {
+    function applyNullability<Type extends GraphQLType | GraphQLInputType>(
+      type: Type,
+      allNullabilities: boolean[]
+    ): Type {
+      const [nullability, ...rest] = allNullabilities;
+      const nullableType = type instanceof GraphQLNonNull ? type.ofType : type;
+      let inner = null;
+      if (nullableType instanceof GraphQLList) {
+        inner = new GraphQLList(applyNullability(nullableType.ofType, rest));
+      } else {
+        if (rest.length > 0) {
+          throw new Error(`Too many booleans passed`);
+        }
+        inner = nullableType;
+      }
+      return nullability
+        ? inner
+        : inner === type
+        ? new GraphQLNonNull(inner)
+        : type; // Optimisation if it's already non-null*/;
```

Comment:
Let's not be too clever here:

```suggestion
        : new GraphQLNonNull(inner);
```

(I'm concerned that if `nullableType` is a `GraphQLList` then `inner` will differ from `type` and thus `type` will be used, whereas it should use `GraphQLNonNull(type)` in this case.)

Reply:
✅ (Actually I copy pasted the previous code that already had the optimisation, that's why)